Modules/AntiBlackout.cs

```diff
@@ -236,6 +258,45 @@ public static void AfterMeetingTasks()
             Logger.Error($"{error}", "AntiBlackout.AfterMeetingTasks");
         }
     }
+    public static void SetRealPlayerRoles()
+    {
+        if (CustomWinnerHolder.WinnerTeam != CustomWinner.Default) return;
+
+        foreach (var ((seerId, targetId), (roletype, _)) in RpcSetRoleReplacer.RoleMap)
+        {
+            // skip host
+            if (seerId == 0) continue;
+            
+            var seer = Utils.GetPlayerById(seerId);
+            var target = Utils.GetPlayerById(targetId);
+            
+            if (seer == null || target == null) continue;
+            if (seer.IsModClient()) continue;
+
+            var realRoleType = roletype;
+            if (target.Data.IsDead)
+            {
+                realRoleType = target.CanUseSabotage() ? RoleTypes.ImpostorGhost : RoleTypes.CrewmateGhost;
+            }
```

Comment:
Note to self: This part of the code can be improved and immediately assign the Guardian Angel role there if necessary